File: lab01/src/problem2_2_6/SystemOfFirstDegreeEquations.java
package problem2_2_6;

import javax.swing.*;

class SystemOfFirstDegreeEquations {
    private double a11, a12, b1, a21, a22, b2;

    public double getA11() {
        return a11;
    }
    public double getA12() {
        return a12;
    }
    public double getB1() {
        return b1;
    }
    public double getA21() {
        return a21;
    }
    public double getA22() {
        return a22;
    }
    public double getB2() {
        return b2;
    }

    public void setA11() {
        this.a11 = Double.parseDouble(JOptionPane.showInputDialog("Ho Tuan Anh 20226100 a11 = "));
    }
    public void setA12() {
        this.a12 = Double.parseDouble(JOptionPane.showInputDialog("Ho Tuan Anh 20226100 a12 = "));
    }
    public void setB1() {
        this.b1 = Double.parseDouble(JOptionPane.showInputDialog("Ho Tuan Anh 20226100 b1 = "));
    }
    public void setA21() {
        this.a21 = Double.parseDouble(JOptionPane.showInputDialog("Ho Tuan Anh 20226100 a21 = "));
    }
    public void setA22() {
        this.a22 = Double.parseDouble(JOptionPane.showInputDialog("Ho Tuan Anh 20226100 a22 = "));
    }
    public void setB2() {
        this.b2 = Double.parseDouble(JOptionPane.showInputDialog("Ho Tuan Anh 20226100 b2 = "));
    }
   
    public void solve() {
        double D = a11 * a22 - a12 * a21, Dx = b1 * a22 - b2 * a12, Dy = a11 * b2 - a21 * b1;
        if (D == 0) {
            if (Dx == 0 && Dy == 0) {
                JOptionPane.showMessageDialog(null, "Ho Tuan Anh 20226100 The system has infinitely many solutions.");
            } else {
                JOptionPane.showMessageDialog(null, "Ho Tuan Anh 20226100 The system has no solution.");
            }
        } else {
            double x = Dx / D;
            double y = Dy / D;
            JOptionPane.showMessageDialog(null, String.format("Ho Tuan Anh 20226100 The solutions are: x = %.2f and y = %.2f", x, y));
        }
    }

}

File: lab01/src/com/oop/problem2_2_6/impl/SystemOfFirstDegreeEquations.kt
package com.oop.problem2_2_6.impl

import com.oop.problem2_2_6.Equation

class SystemOfFirstDegreeEquations(val a11: Double, val a12: Double, val b1: Double,
                                   val a21: Double, val a22: Double, val b2: Double) : Equation {

    override fun display(): String {
        return """
            |System of equations:
            | ._ ${a11}x + ${a12}y = $b1
            | |_ ${a21}x + ${a22}y = $b2
        """.trimMargin()
    }

    override fun solve(): Pair<Any, Any> {
        val d = a11 * a22 - a12 * a21
        val dx = b1 * a22 - b2 * a12
        val dy = a11 * b2 - a21 * b1
        if (d == 0.0) {
            if (dx == 0.0 && dy == 0.0) {
                if (a12 == 0.0 && a22 == 0.0) {
                    if (a11 == 0.0 && a21 == 0.0) {
                        if (b1 == 0.0 && b2 == 0.0) {
                            return Pair("any", "any")
                        } else {
                            throw ArithmeticException("The system has no solution.")
                        }
                    } else if (a11 != 0.0 ) {
                        val y = b1 / a11
                        if (y  * a21 == b2) {
                            return Pair("any", y)
                        } else {
                            throw ArithmeticException("The system has no solution.")
                        }
                    } else if (a21 != 0.0) {
                        val y = b2 / a21
                        if (y * a11 == b1) {
                            return Pair("any", y)
                        } else {
                            throw ArithmeticException("The system has no solution.")
                        }
                    }
                } else if (a11 == 0.0 && a21 == 0.0) {
                    if (a12 != 0.0) {
                        val x = b1 / a12
                        if (x * a22 == b2) {
                            return Pair(x, "any")
                        } else {
                            throw ArithmeticException("The system has no solution.")
                        }
                    } else if (a22 != 0.0) {
                        val x = b2 / a22
                        if (x * a12 == b1) {
                            return Pair(x, "any")
                        } else {
                            throw ArithmeticException("The system has no solution.")
                        }
                    }
                }
            } else {
                throw ArithmeticException("The system has no solution.")
            }
        }
        val x = dx / d
        val y = dy / d
        return Pair(x, y)
    }

}